tests/test_unit/test_cdf_tk/test_utils/test_table_writers.py

```diff
@@ -0,0 +1,100 @@
+from pathlib import Path
+
+import pyarrow.parquet as pq
+import pytest
+
+from cognite_toolkit._cdf_tk.utils.table_writers import (
+    CSVWriter,
+    ParquetWriter,
+    Schema,
+    SchemaColumn,
+    YAMLWriter,
+)
+
+
+@pytest.fixture()
+def example_schema() -> Schema:
+    return Schema(
+        display_name="Example",
+        folder_name="example",
+        kind="Example",
+        format_="csv",
+        columns=[
+            SchemaColumn(name="column1", type="string"),
+            SchemaColumn(name="column2", type="integer"),
+            SchemaColumn(name="column3", type="float"),
+            SchemaColumn(name="column4", type="string", is_array=True),
+        ],
+    )
+
+
+class TestTableFileWriter:
+    def test_write_csv(self, example_schema: Schema, tmp_path: Path) -> None:
+        output_dir = tmp_path / "output"
+
+        with CSVWriter(example_schema, output_dir) as writer:
+            writer.write_rows([("group1", [{"column1": "value1", "column2": 1, "column3": 1.0, "column4": ["value"]}])])
+            writer.write_rows([("group1", [{"column1": "value2", "column3": 2.0}])])
+
+        csv_file = list(output_dir.rglob("*.csv"))
+        assert len(csv_file) == 1
+        actual_file = csv_file[0].read_text()
+        assert actual_file == ("column1,column2,column3,column4\nvalue1,1,1.0,['value']\nvalue2,,2.0,\n")
+
+    def test_write_yaml(self, example_schema: Schema, tmp_path: Path) -> None:
```

Comment:
Should verify that we handle corner cases correctly for yaml, like `false` vs `"false"`. Do we support yaml 1.1 parsers or only >1.2? Older parsers might treat `yes/no/on/off/y/n` etc. as a boolean, so it needs quotes in more cases.